cognite/neat/_session/_show.py

```diff
@@ -14,26 +18,147 @@ class ShowAPI:
+            ?s ?p ?o .
+            FILTER(isIRI(?o))  # Example filter to check if ?o is an IRI (object type)
+            FILTER(BOUND(?o))
+            FILTER(?p != rdf:type)
+
+            ?s a ?ts .
+            ?o a ?to .
+        }
+        LIMIT 100
+        """
+
+        G = nx.DiGraph()
+
+        types = set()
+
+        for (
+            s,
+            p,
+            o,
+            s_type,
+            o_type,
+        ) in self._state.store.graph.query(query):
+            s = remove_namespace_from_uri(s)
+            p = remove_namespace_from_uri(p)
+            o = remove_namespace_from_uri(o)
+            s_type = remove_namespace_from_uri(s_type)
+            o_type = remove_namespace_from_uri(o_type)
+
+            G.add_node(s, label=s, type=s_type)
+            G.add_node(o, label=o, type=o_type)
+            G.add_edge(s, o, label=p)
+
+            types.add(s_type)
+            types.add(o_type)
+
+        return G, types
+
+    def _generate_cytoscape_widget_style(self, types: list[str]) -> list[dict]:
+        widget_style = [
+            {
+                "selector": "edge",
+                "style": {
+                    "width": 1,
+                    "target-arrow-shape": "triangle",
+                    "curve-style": "bezier",
+                    "label": "data(label)",
+                    "font-size": "8px",
+                    "line-color": "black",
+                    "target-arrow-color": "black",
+                },
+            },
+        ]
+
+        colors = self._generate_hex_colors(len(types))
+
+        for i, type_ in enumerate(types):
+            widget_style.append(self._generate_node_cytoscape_style(type_, colors[i]))
+
+        return widget_style
+
+    @staticmethod
+    def _generate_hex_colors(n: int) -> list[str]:
+        """Generate a list of N random HEX color codes."""
+        hex_colors = []
+        for _ in range(n):
+            color = f"#{random.randint(0, 0xFFFFFF):06x}"
+            hex_colors.append(color)
+        return hex_colors
+
+    @staticmethod
+    def _generate_node_cytoscape_style(type_: str, color: str) -> dict:
+        template = {
+            "css": {
+                "content": "data(label)",
+                "text-valign": "center",
+                "color": "black",
+                "font-size": "10px",
+                "width": "mapData(score, 0, 1, 10, 50)",
+                "height": "mapData(score, 0, 1, 10, 50)",
+            },
+        }
+
+        template["selector"] = f'node[type = "{type_}"]'  # type: ignore
+        template["css"]["background-color"] = color
+
+        return template
 
 
 class ShowDataModelAPI:
     def __init__(self, state: SessionState) -> None:
         self._state = state
 
     def __call__(self) -> Any:
+        if not self._state.last_verified_dms_rules and not self._state.last_verified_information_rules:
+            print("No rules have been verified yet.")
+            return None
```

Comment:
Lets discuss this one, I think there is a pattern were we just raise NeatSession errors that are picked up and displayed nicely.